src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs

```diff
@@ -117,7 +118,19 @@ private static IEnumerable<PSRoleAssignment> ToPSRoleAssignments(this IEnumerabl
 
             List<string> objectIds = new List<string>();
             objectIds.AddRange(assignments.Select(r => r.PrincipalId.ToString()));
-            List<PSADObject> adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            List<PSADObject> adObjects = null;
+            try
+            {
+                adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            }
+            catch (CloudException ce) {
+                if (ce.Response.StatusCode == HttpStatusCode.Unauthorized && ce.Error.Code.Equals("Authorization_RequestDenied",StringComparison.OrdinalIgnoreCase))
```

Comment:
@darshanhs90 nit: should we make the "Authorization_RequestDenied` string a constant?

Reply:
Done